### PR TITLE
Add trial panel

### PR DESCRIPTION
**Pull request by VasilyStrelyaev**

### CODE REVIEW

**Comment by VasilyStrelyaev** [comment]
@timbset please check if there are any CSP violations introduced

File: packages/devextreme/js/__internal/core/license/license_validation.ts [hunk 1]
```diff
@@ -139,13 +158,63 @@ function getLicenseCheckParams({ licenseKey, version }: {
   }
 }
 
+export function showTrialPanel(buyNowUrl: string, version: string): void {
```
Comment (IlyaKhD) [inline packages/devextreme/js/__internal/core/license/license_validation.ts]: What about moving this to the trial_panel module?

Comment (IlyaKhD) [reply packages/devextreme/js/__internal/core/license/license_validation.ts]: Much better import statemnt in this case:

```ts
import {
  showTrialPanel,
} from './trial_panel';
```

File: packages/devextreme/js/__internal/core/license/license_validation.ts [hunk 2]
```diff
@@ -139,13 +158,63 @@ function getLicenseCheckParams({ licenseKey, version }: {
...
+  if (!error || internal) {
+    return false;
+  }
+
+  if (preview) {
+    const { major, minor, patch } = parseVersion(version);
+
+    const previousMajor = (
+      minor === 1 ? [major - 1, MAX_MINOR_VERSION, patch] : [major, minor - 1, patch]
+    ).join(SPLITTER);
```
Comment (IlyaKhD) [inline packages/devextreme/js/__internal/core/license/license_validation.ts]: This can be extracted as `getPreviousMajorVersion`

Comment (IlyaKhD) [reply packages/devextreme/js/__internal/core/license/license_validation.ts]: Probably `parseVersion` `getPreviousMajorVersion` and `ParsedVersion` (better simply `Version`) should be moved to a separate module (like 'core/utils/version')

**Review by timbset — APPROVED** [review]

**Review by IlyaKhD — APPROVED** [review]

**Pull request merged**